packages/recorder/src/index.ts

```diff
@@ -3,21 +3,59 @@ import {SessionManager} from './session_recording/sessions_manager';
 import {initTerminal} from './terminal_interaction';
 import {AppDependencies} from './types';
 
-const defaultDestination = process.env.DESTINATION;
-const useDestinationHostHeader = process.env.USE_DESTINATION_HOST_HEADER;
+import {Command} from 'commander';
+const program = new Command();
 
-if (!defaultDestination && useDestinationHostHeader !== 'true') {
-    console.log('Please provide a destination via the DESTINATION enviornment variable, or bypass this check by setting the USE_DESTINATION_HOST_HEADER bool environment variable');
+const runServerForDestination = async (host: string, port: string, deps: AppDependencies) => {
+    const app = initApp({
+        ...deps,
+        defaultDestination: host,
+    });
+
+    return new Promise<void>(r => {
+        app.listen(port, () => {
+            console.log(`http://localhost:${port} -> ${host}`);
+            setTimeout(() => {
+                r();
+            }, 50);
+        });
+    });
 }
 
-const sessionManager = new SessionManager();
-const deps: AppDependencies = {sessionManager, defaultDestination};
+program
+    .name('fc-record')
+    .description('CLI to record HTTP request sessions')
+    .version('0.0.1');
+
+// Usage: Destination host and local port pairs. A vertical bar is used to join the destination host and local port. The destinations are separated by spaces.
+program.command('record')
+    .description('Record requests')
+    .option('-d, --destinations [host|port...]', 'Destination/port mappings to proxy and record requests')
+    .option('-h, --includeHeaders', 'Include HTTP headers in output', false)
+    .action(async ({destinations, includeHeaders}: {destinations: string[], includeHeaders: boolean}, options) => {
+        console.log(destinations, '\n');
+
+        const sessionManager = new SessionManager();
+        const deps: AppDependencies = {
+            sessionManager,
+            defaultDestination: '',
+            includeHeaders,
+        };
+
+        for (const dest of destinations) {
+            const [host, port] = dest.split('|');
+            await runServerForDestination(host, port, deps);
+        }
```

Comment:
TODO: Harden this code. Should make sure we received the CLI arg, and validate the format of the arg before using